pkg/restore/merge_service_account_test.go

```diff
@@ -0,0 +1,701 @@
+						{ "name": "my-old-secret" },
+						{ "name": "secrete"}
+					]
+				}`,
+			),
+		},
+		{
+			name: "service accounts with labels and annotations",
+			fromCluster: unstructuredOrDie(
+				`{
+					"apiVersion": "v1",
+					"kind": "ServiceAccount",
+					"metadata": {
+						"namespace": "ns1",
+						"name": "default",
+						"labels": {
+							"l1": "v1",
+							"l2": "v2",
+							"l3": "v3"
+						},
+						"annotations": {
+							"a1": "v1",
+							"a2": "v2",
+							"a3": "v3",
+							"a4": "v4"
+						}
+					},
+					"secrets": [
+						{ "name": "default-token-abcde" }
+					]
+				}`,
+			),
+
+			fromBackup: unstructuredOrDie(
+				`{
+					"kind": "ServiceAccount",
+					"apiVersion": "v1",
+					"metadata": {
+						"namespace": "ns1",
+						"name": "default",
+						"labels": {
+							"l1": "v1",
+							"l2": "v2",
+							"l3": "v3",
+							"l4": "v4",
+							"l5": "v5"
+						},
+						"annotations": {
+							"a1": "v1",
+							"a2": "v2",
+							"a3": "v3",
+							"a4": "v4",
+							"a5": "v5",
+							"a6": "v6"
+						}
+					},
+					"secrets": [
+						{ "name": "default-token-xzy12" }
+					]
+				}`,
+			),
+		},
+	}
+
+	var desired *unstructured.Unstructured
+
+	for _, test := range tests {
+		b.Run(test.name, func(b *testing.B) {
+			for n := 0; n < b.N; n++ {
+				desired, _ = mergeServiceAccounts(test.fromCluster, test.fromBackup)
+			}
+
+			mergedServiceAccountsBenchmarkResult = desired
+		})
+	}
+}
+
+func TestMergeLocalObjectReferenceSlices(t *testing.T) {
+	tests := []struct {
+		name     string
+		first    []corev1api.LocalObjectReference
+		second   []corev1api.LocalObjectReference
+		expected []corev1api.LocalObjectReference
+	}{
+		{
+			name: "two slices without overlapping elements",
+			first: []corev1api.LocalObjectReference{
+				corev1api.LocalObjectReference{Name: "lor1"},
+				corev1api.LocalObjectReference{Name: "lor2"},
+			},
+			second: []corev1api.LocalObjectReference{
+				corev1api.LocalObjectReference{Name: "lor3"},
+				corev1api.LocalObjectReference{Name: "lor4"},
+			},
+			expected: []corev1api.LocalObjectReference{
+				corev1api.LocalObjectReference{Name: "lor1"},
+				corev1api.LocalObjectReference{Name: "lor2"},
+				corev1api.LocalObjectReference{Name: "lor3"},
+				corev1api.LocalObjectReference{Name: "lor4"},
+			},
+		},
+		{
+			name: "two slices with an overlapping element",
+			first: []corev1api.LocalObjectReference{
+				corev1api.LocalObjectReference{Name: "lor1"},
+				corev1api.LocalObjectReference{Name: "lor2"},
+			},
+			second: []corev1api.LocalObjectReference{
+				corev1api.LocalObjectReference{Name: "lor3"},
+				corev1api.LocalObjectReference{Name: "lor2"},
+			},
+			expected: []corev1api.LocalObjectReference{
+				corev1api.LocalObjectReference{Name: "lor1"},
+				corev1api.LocalObjectReference{Name: "lor2"},
+				corev1api.LocalObjectReference{Name: "lor3"},
+			},
+		},
+		{
+			name: "merging always adds elements to the end",
```

Comment:
Just curious if order matters? Would it make more sense to sort the 2 slices before comparing them?

Reply:
I don't *think* it does; I haven't delved into the k8s code that works on the references for the (local) object references to know. This test is perhaps mostly about demonstrating current behavior.